src/zenml/artifacts/load_directory_materializer.py

```diff
@@ -0,0 +1,80 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Only-load materializer for directories."""
+
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, ClassVar, Tuple, Type
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class LoaderDirectoryMaterializer(BaseMaterializer):
```

Comment:
If I remember correctly, I implemented a `DirectoryMaterializer` which works to both save and load directories when an artifact was of type `Path`. Any specific reason we do not implement the `save(...)` method for this materializer as well?

I might be wrong here, but because this materializer has the `pathlib.Path` as it's associated type, it will even be registered for this type in the materializer registry in case it gets imported somehow? Which would then break all returns of type `pathlib.Path`.

Reply:
The `save` was not implemented to protect from misuse of this specific materializer by intent. And this implementation echos the `DirectoryMaterializer` you created in projects.

Yes, it works with `pathlib.Path`, shall I create a child class from `pathlib.Path` to protect this case?

Reply:
I see, I think my preferred option would be somehow allowing special materializers without associated types that don't get picked up as default for anything in the registry. I'm not sure however how much effort this is.

Something I think we should do in any case, even if we keep it as-is: The `NotImplementedError` that gets raised in the `save(...)` method should explain that this is a special materializer and should not be used for regular artifacts.

Reply:
I made it through the `PreexistingArtifactPath` derived from the `Path`, otherwise logic of input artifacts blows up. Please check the recent changes again 🙂